src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Windows.cs

```diff
@@ -24,6 +24,18 @@ public partial class MemoryMappedFile
             SafeFileHandle? fileHandle = fileStream != null ? fileStream.SafeFileHandle : null;
             Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = GetSecAttrs(inheritability);
 
+            if (fileStream != null)
+            {
+                if (access == MemoryMappedFileAccess.Read && capacity > fileStream.Length)
+                {
+                    throw new ArgumentException(SR.Argument_ReadAccessWithLargeCapacity);
+                }
+
+                if (access == MemoryMappedFileAccess.Write)
```

Comment:
The `CreateNew` method was throwing exception, which you removed. Now it's going to depend on the `fileStream` being not `null`, which for `CreateNew` is never the case.

Reply:
Touching `CreateNew` was never the intention. I miss that because I only added a test for CreateFromFile with a stream. I removed change to the `CreateNew`  method and I added one more test.